scripts/protoc-swagger-gen.sh

```diff
@@ -2,36 +2,40 @@
 
 set -eo pipefail
 
+SWAGGER_DIR=./swagger-proto
+
+# configure buf workspace settings
+mkdir -p "$SWAGGER_DIR/proto"
+printf "version: v1\ndirectories:\n  - proto\n  - third_party" > "$SWAGGER_DIR/buf.work.yaml"
+printf "version: v1\nname: buf.build/axelarnetwork/axelar-core\n" > "$SWAGGER_DIR/proto/buf.yaml"
+cp ./proto/buf.gen.swagger.yaml "$SWAGGER_DIR/buf.gen.swagger.yaml"
+
+# copy existing proto files
+cp -r ./proto/axelar "$SWAGGER_DIR/proto"
+
+# create temporary folder to store intermediate results from `buf generate`
 mkdir -p ./tmp-swagger-gen
 
-# Get the path of the cosmos-sdk repo from go/pkg/mod
-proto_dirs=$(find ./proto ./third_party/proto -path -prune -o -name '*.proto' -print0 | xargs -0 -n1 dirname | sort | uniq)
+cd "$SWAGGER_DIR"
 
+# Get the path of the cosmos-sdk repo from go/pkg/mod
+proto_dirs=$(find ./proto ./third_party -path -prune -o -name '*.proto' -print0 | xargs -0 -n1 dirname | sort | uniq)
 for dir in $proto_dirs; do
   # generate swagger files (filter query files)
-  proto_files=$(find "${dir}" \( -name 'query.proto' -o -name 'service.proto' \))
+  proto_files=$(find "${dir}" -maxdepth 1 \( -name 'query.proto' -o -name 'service.proto' \))
   if [[ -z "$proto_files" ]]; then
-    continue
+      continue
   fi
 
   for proto_file in ${proto_files}; do
-    buf protoc  \
-    -I "proto" \
-    -I "third_party/proto" \
-    "$proto_file" \
-    --swagger_out=./tmp-swagger-gen \
-    --swagger_opt=logtostderr=true \
-    --swagger_opt=fqn_for_swagger_name=true \
-    --swagger_opt=simple_operation_ids=true
+    buf generate --template buf.gen.swagger.yaml $proto_file
   done
 done
 
-cd ./client/docs
-yarn install
-yarn combine
-yarn convert
-yarn build
-cd ../../
+cd ..
+
+swagger-combine ./client/docs/config.json -o ./client/docs/static/swagger/swagger.yaml -f yaml --continueOnConflictingPaths true --includeDefinitions true
 
 # clean swagger files
-rm -rf ./tmp-swagger-gen
+rm -rf "$SWAGGER_DIR"
+rm -rf ./tmp-swagger-gen
```

Comment:
is this an issue with the generated docs? They all point to a directory that we delete here, don't they?

Reply:
They are temporary working directories for Swagger Gen, and the final output is in `/client/docs/`